scipy/signal/tests/test_fir_filter_design.py

```diff
@@ -383,12 +383,84 @@ def test_hilbert(self):
         Hmag = abs(H)
 
         # should have a zero at 0 and pi (in this case close to zero)
-        assert_((Hmag[[0,-1]] < 0.02).all(), "Zero at zero and pi")
+        assert_((Hmag[[0, -1]] < 0.02).all(), "Zero at zero and pi")
 
         # check that the pass band is close to unity
         idx = (f > a) * (f < 0.5-a)
         assert_((abs(Hmag[idx] - 1) < 0.015).all(), "Pass Band Close To Unity")
 
 
+class TestFirls(TestCase):
+
+    def test_bad_args(self):
+        # even numtaps
+        assert_raises(ValueError, firls, 10, [0.1, 0.2], [0])
+        # odd bands
+        assert_raises(ValueError, firls, 11, [0.1, 0.2, 0.4], [0])
+        # len(bands) != 2*len(desired)
+        assert_raises(ValueError, firls, 11, [0.1, 0.2, 0.3, 0.4], [0])
+        # non-monotonic bands
+        assert_raises(ValueError, firls, 11, [0.2, 0.1], [0])
+        # negative desired
+        assert_raises(ValueError, firls, 11, [0.1, 0.2], [-1])
+        # len(weight) != len(desired)
+        assert_raises(ValueError, firls, 11, [0.1, 0.2], [0], [1, 2])
+        # negative weight
+        assert_raises(ValueError, firls, 11, [0.1, 0.2], [0], [-1])
+
+    def test_firls(self):
+        N = 11  # number of taps in the filter
+        a = 0.1  # width of the transition band
+
+        # design a halfband symmetric low-pass filter
+        h = firls(11, [0, a, 0.5-a, 0.5], [1, 0], fs=1)
+
+        # make sure the filter has correct # of taps
+        assert_equal(len(h), N)
+
+        # make sure it is symmetric
+        midx = (N-1) // 2
+        assert_array_almost_equal(h[:midx], h[:-midx-1:-1])
+
+        # make sure the center tap is 0.5
+        assert_almost_equal(h[midx], 0.5)
+
+        # For halfband symmetric, odd coefficients (except the center)
+        # should be zero (really small)
+        hodd = np.hstack((h[1:midx:2], h[-midx+1::2]))
+        assert_array_almost_equal(hodd, 0)
+
+        # now check the frequency response
+        w, H = freqz(h, 1)
+        f = w/2/np.pi
+        Hmag = np.abs(H)
+
+        # check that the pass band is close to unity
+        idx = (f > 0) * (f < a)
```

Comment:
In the previous PR, Ralf suggested `np.logical_and` here for clarity, which I support